frontend/public/extend/devconsole/components/build-config/BuildConfig.tsx

```diff
@@ -0,0 +1,87 @@
+/*eslint-disable no-unused-vars, no-undef */
+import * as React from 'react';
+import { Checkbox } from './../../../../../public/components/checkbox';
+import { EnvironmentPage } from './../../../../../public/components/environment';
+
+interface NameValueType {
+  name: string;
+  value: string;
+}
+
+interface NameValueFormType {
+  name: string;
+  valueForm: {
+    configMapKeyRef: {
+      key: string;
+      name: string
+    }
+  }
+}
+interface BuildConfigProps {
+  onConfigureWebhookChange: React.ReactEventHandler<HTMLInputElement>;
+  onAutomaticallyBuildChange: React.ReactEventHandler<HTMLInputElement>;
+  onLaunchFirstBuildChange: React.ReactEventHandler<HTMLInputElement>;
+  onEnviromentVariableChange: (obj: NameValueType | NameValueFormType) => void;
```

Comment:
The param is an array of values.
```suggestion
  onEnviromentVariableChange: (envPairs: (NameValueType | NameValueFormType)[]) => void;
```